builtin/providers/azurerm/resource_arm_virtual_machine_scale_set.go

```diff
@@ -0,0 +1,1107 @@
+	var buf bytes.Buffer
+	m := v.(map[string]interface{})
+	buf.WriteString(fmt.Sprintf("%s-", m["publisher"].(string)))
+	buf.WriteString(fmt.Sprintf("%s-", m["offer"].(string)))
+	buf.WriteString(fmt.Sprintf("%s-", m["sku"].(string)))
+	buf.WriteString(fmt.Sprintf("%s-", m["version"].(string)))
+
+	return hashcode.String(buf.String())
+}
+
+func resourceArmVirtualMachineScaleSetSkuHash(v interface{}) int {
+	var buf bytes.Buffer
+	m := v.(map[string]interface{})
+	buf.WriteString(fmt.Sprintf("%s-", m["name"].(string)))
+	if m["tier"] != nil {
+		buf.WriteString(fmt.Sprintf("%s-", m["tier"].(string)))
+	}
+	buf.WriteString(fmt.Sprintf("%d-", m["capacity"].(int)))
+
+	return hashcode.String(buf.String())
+}
+
+func resourceArmVirtualMachineScaleSetStorageProfileOsDiskHash(v interface{}) int {
+	var buf bytes.Buffer
+	m := v.(map[string]interface{})
+	buf.WriteString(fmt.Sprintf("%s-", m["name"].(string)))
+	//buf.WriteString(fmt.Sprintf("%v-", m["caching"].(string)))
+	//buf.WriteString(fmt.Sprintf("%v-", m["create_option"].(string)))
+	if m["image"] != nil {
+		buf.WriteString(fmt.Sprintf("%s-", m["image"].(string)))
+	}
+	if m["os_type"] != nil {
+		buf.WriteString(fmt.Sprintf("%s-", m["os_type"].(string)))
+	}
+
+	return hashcode.String(buf.String())
+}
+
+func resourceArmVirtualMachineScaleSetNetworkConfigurationHash(v interface{}) int {
+	var buf bytes.Buffer
+	m := v.(map[string]interface{})
+	buf.WriteString(fmt.Sprintf("%s-", m["name"].(string)))
+	buf.WriteString(fmt.Sprintf("%t-", m["primary"].(bool)))
+	return hashcode.String(buf.String())
+}
+
+func resourceArmVirtualMachineScaleSetsOsProfileHash(v interface{}) int {
+	var buf bytes.Buffer
+	m := v.(map[string]interface{})
+	buf.WriteString(fmt.Sprintf("%s-", m["computer_name_prefix"].(string)))
+	buf.WriteString(fmt.Sprintf("%s-", m["admin_username"].(string)))
+	if m["custom_data"] != nil {
+		buf.WriteString(fmt.Sprintf("%s-", m["custom_data"].(string)))
+	}
+	return hashcode.String(buf.String())
+}
+
+func resourceArmVirtualMachineScaleSetOsProfileLinuxConfigHash(v interface{}) int {
+	var buf bytes.Buffer
+	m := v.(map[string]interface{})
+	buf.WriteString(fmt.Sprintf("%t-", m["disable_password_authentication"].(bool)))
+
+	return hashcode.String(buf.String())
+}
+
+func resourceArmVirtualMachineScaleSetOsProfileLWindowsConfigHash(v interface{}) int {
+	var buf bytes.Buffer
+	m := v.(map[string]interface{})
+	if m["provision_vm_agent"] != nil {
+		buf.WriteString(fmt.Sprintf("%t-", m["provision_vm_agent"].(bool)))
+	}
+	if m["enable_automatic_upgrades"] != nil {
+		buf.WriteString(fmt.Sprintf("%t-", m["enable_automatic_upgrades"].(bool)))
+	}
+	return hashcode.String(buf.String())
+}
+
+func expandVirtualMachineScaleSetSku(d *schema.ResourceData) (*compute.Sku, error) {
+	skuConfig := d.Get("sku").(*schema.Set).List()
+
+	if len(skuConfig) != 1 {
+		return nil, fmt.Errorf("Cannot specify more than one SKU.")
+	}
```

Comment:
We have helper/schema support for this now!

Check out `MaxItems` - https://github.com/hashicorp/terraform/blob/d175b0418227d1d578f7dd8c9d8c68a398ba731a/helper/schema/schema.go#L103-L108